user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx

```diff
@@ -45,26 +46,50 @@ export function DropdownMenu(props: DropdownMenuProps) {
       handleToggleExpand();
       setFocus(true);
     } else {
-      const firstItem = document.querySelector(`#menu-link-${id}-0`);
-      const lastItem = document.querySelector(`#menu-link-${id}-${submenuItemCount - 1}`);
+      const firstItem: HTMLLinkElement | null = document.querySelector(`#menu-link-${id}-0`);
+      const lastItem: HTMLLinkElement | null = document.querySelector(
+        `#menu-link-${id}-${submenuItemCount - 1}`,
+      );
       if (ev.key === 'ArrowDown') {
         if (ev.currentTarget === lastItem) {
-          handleToggleExpand();
-          setFocus(true);
+          if (firstItem) firstItem.focus();
+          ev.preventDefault();
         } else {
           const nextItem = ev.currentTarget.parentElement?.nextElementSibling;
           if (nextItem) (nextItem.children[0] as HTMLLIElement).focus();
         }
+        ev.preventDefault();
       } else if (ev.key === 'ArrowUp') {
         if (ev.currentTarget === firstItem) {
-          handleToggleExpand();
-          setFocus(true);
+          if (lastItem) lastItem.focus();
+          ev.preventDefault();
         } else {
           const previousItem = ev.currentTarget.parentElement?.previousElementSibling;
           if (previousItem) (previousItem.children[0] as HTMLLIElement).focus();
         }
-      } else if (ev.key === 'Tab' && ev.shiftKey === false) {
+        ev.preventDefault();
```

Comment:
It seems like this makes line 65 unnecessary, or applies to the else block on 66 where maybe we don't want to.